ambu

```diff
@@ -1,22 +1,25 @@
 #!/bin/bash
 
 MAKEMODE=$1
-CURRENTDIR=`pwd`
+CURRENTDIR=${2:-`pwd`}
 PROJECT=`basename $CURRENTDIR`
 PARENTDIR=`dirname $CURRENTDIR`
 BUILDDIR=$PARENTDIR/".build_"$PROJECT
 PROJECTNAME=$PROJECT".ino"
 
-usage()
-{
-cat << EOF
-usage: $0 [makefile target]
-(see Arduino Makefile for target details)
-EOF
+usage() {
+    echo "usage: $0 target [project dir]"
+    echo "(see Arduino Makefile for target details)"
 }
 
 if [ -z "$1" ]; then
     usage
+    exit 0
+fi
+
+if [ ! -d "$CURRENTDIR" ]; then
+    echo "Project given in argument is not a directory or directory"
```

Comment:
@Adman Two things here: 

1. It would help a lot if the message itself had the path/directory in it.
2. I believe that saying that the argument is not a directory is sufficient. If it does not exist it cannot be a directory -- saying it again does not add any additional informational value.